model/metrics/gen-ai.yaml

```
@@ -0,0 +1,47 @@
+groups:
+  - id: metric_attributes.gen_ai
+    type: attribute_group
+    brief: 'This group describes GenAI metrics attributes'
+    attributes:
+      - ref: server.address
+        requirement_level: recommended
+      - ref: server.port
+        requirement_level:
+          conditionally_required: If `sever.address` is set.
+      - ref: gen_ai.response.model
+        requirement_level: recommended
+      - ref: gen_ai.request.model
+        requirement_level: required
+      - ref: gen_ai.system
+        requirement_level: required
+      - ref: gen_ai.operation.name
+        requirement_level: required
+  - id: metric.gen_ai.client.token.usage
+    type: metric
+    metric_name: gen_ai.client.token.usage
+    brief: 'Measures number of input and output tokens used'
+    instrument: histogram
+    unit: "{token}"
+    stability: experimental
+    extends: metric_attributes.gen_ai
+    attributes:
+      - ref: gen_ai.token.type
+        requirement_level: required
+  - id: metric.gen_ai.client.operation.duration
+    type: metric
+    metric_name: gen_ai.client.operation.duration
+    brief: 'GenAI operation duration'
+    instrument: histogram
+    unit: "s"
+    stability: experimental
+    extends: metric_attributes.gen_ai
+    attributes:
+      - ref: error.type
+        requirement_level:
+          conditionally_required: "if the operation ended in an error"
+        note: |
+          The cardinality of `error.type` SHOULD be low.
+
+          When working across multiple models, it is RECOMMENDED to use a common set of error types.
+
+          Additional details may be captured in domain-specific attributes.
```

```suggestion
          The `error.type` SHOULD match the error code returned by the Generative AI provider or the client library,
          the canonical name of exception that occurred, or another low-cardinality error identifier.
          Instrumentations SHOULD document the list of errors they report.
```

We don't attempt to normalize or standardize error types (attempted many times and failed every time)